odc/stats/plugins/l34_utils/lc_level3.py

```diff
@@ -0,0 +1,50 @@
+import xarray as xr
+
+from odc.stats._algebra import expr_eval
+
+NODATA = 255
+
+
+def lc_level3(xx: xr.Dataset):
+
+    # Cultivated pipeline applies a mask which feeds only terrestrial veg (110) to the model
+    # Just exclude no data (255) and apply the cultivated results
+    res = expr_eval(
+        "where(a<nodata, a, b)",
+        {"a": xx.cultivated_class.data, "b": xx.classes_l3_l4.data},
+        name="mask_cultivated",
+        dtype="float32",
+        **{"nodata": NODATA},
+    )
+
+    # Mask urban results with bare sfc (210)
+    res = expr_eval(
+        "where(a==_u, b, a)",
+        {
+            "a": res,
+            "b": xx.urban_classes.data,
+        },
+        name="mark_urban",
+        dtype="uint8",
+        **{"_u": 210},
+    )
+
+    intertidal_mask = expr_eval(
```

Comment:
this mask is redundant.

Reply:
This mask is used in level4 water classification.

Reply:
I mean it's redundant in a sense that l3 and l4 results doesn't have to be generated in that order. The water persistence and consequently l4 water classification can happen well ahead of l3 aggregation.  https://github.com/opendatacube/odc-stats/blob/40ab39861f9cec261360568bdc01807dc8ba0c7c/odc/stats/plugins/lc_level34.py#L92 and https://github.com/opendatacube/odc-stats/blob/40ab39861f9cec261360568bdc01807dc8ba0c7c/odc/stats/plugins/lc_level34.py#L107 doesn't depend on the aggregated results of l3, so l3 aggregation needs not to happen first.

Reply:
This is addressed now.